src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -354,6 +356,10 @@ public Index GetIndex(string name)
         {
             if (definition == null)
                 throw new ArgumentNullException(nameof(definition));
+
+            if (_documentDatabase.Configuration.Core.FeaturesAvailability == FeaturesAvailability.Stable && definition.Type.IsJavaScript())
+                throw new IndexCreationException($"Could not create index '{definition.Name}'. Database does not support 'JavaScript' indexes. Please enable experimental features by changing '{RavenConfiguration.GetKey(x => x.Core.FeaturesAvailability)}' configuration value to '{nameof(FeaturesAvailability.Experimental)}'.");
```

Comment:
Better add a common method `AssertExperimentalFeature(featureName)`, so we can use it in the future.